rosbag2_transport/test/rosbag2_transport/test_record_all.cpp

```diff
@@ -159,10 +159,10 @@ TEST_F(RecordIntegrationTestFixture, published_messages_from_topic_and_service_a
   auto & writer = recorder->get_writer_handle();
   auto & mock_writer = dynamic_cast<MockSequentialWriter &>(writer.get_implementation_handle());
 
-  size_t expected_messages = 3;
+  constexpr size_t expected_messages = 3;
   auto ret = rosbag2_test_common::wait_until_shutdown(
     std::chrono::seconds(5),
-    [&mock_writer, &expected_messages]() {
+    [&mock_writer]() {
```

Comment:
@clalancette Ahh.. This trick doesn't work for Windows.
See CI job failure messages
```

C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\test\rosbag2_transport\test_record_all.cpp(72,1): error C3493: 'expected_messages' cannot be implicitly captured because no default capture mode has been specified [C:\ci\ws\build\rosbag2_transport\test_record_all.vcxproj]

C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\test\rosbag2_transport\test_record_all.cpp(122,1): error C3493: 'expected_messages' cannot be implicitly captured because no default capture mode has been specified [C:\ci\ws\build\rosbag2_transport\test_record_all.vcxproj]

C:\ci\ws\src\ros2\rosbag2\rosbag2_transport\test\rosbag2_transport\test_record_all.cpp(166,1): error C3493: 'expected_messages' cannot be implicitly captured because no default capture mode has been specified [C:\ci\ws\build\rosbag2_transport\test_record_all.vcxproj]
```